src/style/style_layer/fill_extrusion_style_layer.js

```diff
@@ -77,9 +77,20 @@ function getIntersectionDistance(projectedQueryGeometry: Array<Point>, projected
         // triangle of the face, using only the xy plane. It doesn't matter if the
         // point is outside the first triangle because all the triangles in the face
         // are in the same plane.
-        const a = projectedFace[0];
-        const b = projectedFace[1];
-        const c = projectedFace[3];
+        //
+        // Check whether points are coincident and use other points if they are.
+        let i = 0;
+        const a = projectedFace[i++];
+        let b, c;
+        while (!b || a.equals(b)) {
+            b = projectedFace[i++];
+            if (!b) return Infinity;
+        }
+        while (!c || a.equals(c) || b.equals(c)) {
+            c = projectedFace[i++];
+            if (!c) return Infinity;
+        }
```

Comment:
Looks good to me, just one question.
Ss numerical stability a concern with with floating point rounding errors with the `equals` checks? Would doing a distance check with a small epsilon be safer?

Reply:
Hmm... good question. I think the minimum distance (when not equal) should be safely large because vector tile positions are stored as (fairly small) integers. Also, any precision issues would just result in sorting errors which wouldn't be the end of the world.

I think it makes sense to leave it as is for now.